Lazy load node-forge only if needed (Make Parcel ~300KB smaller)

aminya opened the pull request:
# ↪️ Pull Request

This changes the usage of `generateCertificate` so it is only loaded when a HTTPS server is going to be created. Node-forge which is used by `generateCertificate` is ~300KB or more which is now lazy loaded.

<!---
Provide a general summary of the pull request here
Please look for any issues that this PR resolves and tag them in the PR.
-->

## 🚨 Test instructions

The current server/hmr integration tests cover this.
<!-- In case it is impossible (or too hard) to reliably test this feature/fix with unit tests, please provide test instructions! -->

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

aminya committed:
Only import what is needed from node-forge

aminya committed:
Lazy load node-forge only if needed

aminya committed:
Do not export generateCertificate from utils

mischnic commented:
This should make **no** difference because we use the lazy mode of Babel's esm-to-cjs transform https://github.com/parcel-bundler/parcel/blob/cb9b82e3693c9c64dbb3aba1b420308d294d2407/packages/dev/babel-preset/index.js#L22

aminya commented:
If we can find a way to remove node-forge altogether, it would make the code much smaller.

aminya commented:
Not sure why the tests fail.

aminya committed:
Fix generateCertificate dynamic import

aminya commented:
OK. I fixed the tests. The dynamic import is surprisingly wrapping the exports of the module in one another object. Not sure if this is caused by Flow or Babel. So I had to write it like this:
```js
const generateCertificate = (await import('./generateCertificate.js'))
      .default.default;
```

aminya committed:
Do not export getCertificate which is only used inside http-server

aminya commented:
It seems that Flow doesn't see this extra `default`. I am guessing this is added by Babel.

aminya committed:
Ignore the Flow error for the dynamic import

aminya commented:
> This should make **no** difference because we use the lazy mode of Babel's esm-to-cjs transform

Using Node's built-in async `import` is better than using a Babel shim which wraps sync `require`. Also, in #6194, `node-forge` is added to the bundle. Parcel only does splitting whenever dynamic imports are used.

aminya committed:
Fallback to the imported object itself

The root cause of this strange behavior is not known, so we should fallback to the object itself in case the test-env is different from the production-env
